src/Examples/UnitModels/Operations/skeleton_unit_testing.ipynb

```diff
@@ -0,0 +1,860 @@
+      "Separation factor:  1037.618814931264\n",
+      "Condensation duty:  5.812711152340449 kW\n",
+      "Duty per mole water recovered:  0.011324013423297993  kW-h / mol\n"
+     ]
+    }
+   ],
+   "source": [
+    "# separation factor for results analysis - x is inlet water fraction, y is permeate water fraction\n",
+    "m.fs.x = Expression(expr=(m.fs.pervap.inlet.flow_mol_phase_comp[0, \"Liq\", \"water\"] /\n",
+    "                           sum(m.fs.pervap.inlet.flow_mol_phase_comp[0, \"Liq\", i] for i in m.fs.pervap.comp_list)\n",
+    "                          ))\n",
+    "m.fs.y = Expression(expr=(m.fs.pervap.permeate.flow_mol_phase_comp[0, \"Liq\", \"water\"] /\n",
+    "                           sum(m.fs.pervap.permeate.flow_mol_phase_comp[0, \"Liq\", i] for i in m.fs.pervap.comp_list)\n",
+    "                          ))\n",
+    "m.fs.separation_factor = Expression(expr=(m.fs.y/(1-m.fs.y)) / (m.fs.x/(1-m.fs.x)))\n",
+    "\n",
+    "print('Inlet water mole fraction: ', value(m.fs.x))\n",
+    "print('Permeate water mole fraction: ', value(m.fs.y))\n",
+    "print('Separation factor: ', value(m.fs.separation_factor))\n",
+    "print('Condensation duty: ', value(m.fs.pervap.cond_duty[0]/1000), 'kW')\n",
+    "print('Duty per mole water recovered: ', value(\n",
+    "    m.fs.pervap.cond_duty[0]/(1000*m.fs.pervap.permeate.flow_mol_phase_comp[0, \"Liq\", \"water\"]*3600)),' kW-h / mol')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# check results\n",
+    "assert check_optimal_termination(results)\n",
+    "assert_units_consistent(m)\n",
+    "\n",
+    "assert value(m.fs.pervap.permeate.flow_mol_phase_comp[0, \"Liq\", \"water\"]) == pytest.approx(0.1426, rel=1e-3)\n",
+    "assert value(m.fs.pervap.permeate.flow_mol_phase_comp[0, \"Liq\", \"ethylene_glycol\"]) == pytest.approx(0.0002667, rel=1e-3)\n",
+    "assert value(m.fs.pervap.retentate.flow_mol_phase_comp[0, \"Liq\", \"water\"]) == pytest.approx(0.1974, rel=1e-3)\n",
+    "assert value(m.fs.pervap.retentate.flow_mol_phase_comp[0, \"Liq\", \"ethylene_glycol\"]) == pytest.approx(0.6597, rel=1e-3)\n",
+    "assert value(m.fs.separation_factor) == pytest.approx(1038, rel=1e-3)\n",
+    "assert value(m.fs.pervap.cond_duty[0]) == pytest.approx(5813, rel=1e-3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 3. Optimization\n",
+    "\n",
+    "Suppose we wish to characterize the membrane behavior by calculating the maximum inlet water mole fraction allowing a separation factor of at least 100 (typical value for high-efficiency separation processes such as gas separation of CO2/N2). We need to add some lower bounds on the flow variables to ensure physically-sound solutions; alternatively, we could have specified these bounds while defining the variables above. We can quickly modify and resolve the model, and check some key results:"
```

Comment:
You mentioned bounds in the description, but the code doesn't set bounds on any variables.